compiler/sem.nim

```diff
@@ -671,8 +671,9 @@ proc defaultNodeField(c: PContext, a: PNode, aTyp: PType, checkDefault: bool): P
     if child != nil:
       let node = newNode(nkIntLit)
       node.intVal = toInt64(lengthOrd(c.graph.config, aTypSkip))
-      result = semExpr(c, newTree(nkCall, newSymNode(getSysSym(c.graph, a.info, "arrayWith"), a.info),
-              semExprWithType(c, child),
+      let typeNode = newNode(nkSym)
```

Comment:
You should be able to make an `nkType` node with a `typedesc` type as is done [here](https://github.com/nim-lang/Nim/blob/46bb47a444bd377860d832fc1c62b262343f36a2/compiler/sem.nim#L608-L613) instead.